compiler/src/emit.rs

```diff
@@ -0,0 +1,106 @@
+use analyzer::relations::Symbol;
+use analyzer::types::hir::{Declaration, ExprKind, TypedExpr};
+use analyzer::types::*;
+use ast::value::LiteralValue;
+
+use crate::bytecode::{Bytecode, Opcode};
+use crate::constant_pool::ConstantPool;
+use crate::emit::invoke::emit_process_call;
+use crate::emit::jump::{emit_break, emit_conditional, emit_continue, emit_loop};
+
+mod invoke;
+mod jump;
+
+pub struct EmissionState {
```

Comment:
```suggestion
#[derive(Debug, Clone, Default)]
pub struct EmissionState {
```
Derive the `Default` trait instead of implementing a custom `new` constructor.